ironfish/src/memPool/recentlyEvictedCache.ts

```diff
@@ -0,0 +1,218 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Logger } from '../logger'
+import { MetricsMonitor } from '../metrics'
+import { TransactionHash } from '../primitives/transaction'
+import { PriorityQueue } from './priorityQueue'
+
+interface EvictionQueueEntry {
+  hash: TransactionHash
+  feeRate: bigint
+}
+
+interface ExpireAtQueueEntry {
+  hash: TransactionHash
+  expireAtSequence: number
+}
+
+/**
+ * A cache to track transactions that have recently been evicted from the mempool.
+ *
+ * This is primarily a mechanism to prevent transactions from being re-added to the mempool
+ * within a reasonable duration if for some reason they are evicted (i.e. the mempool is full
+ * and the transaction is underpriced).
+ *
+ * When full, transactions are evicted in order of `feeRate` descending to make room for the
+ * new transaction. This order is chosen because these transactions most likely to be successfully
+ * re-introduced into the mempool
```

Comment:
This is a little unclear -- I'd say something like "When the RecentlyEvictedCache is full, the transaction with the lowest feeRate is removed to make room for the new transaction. This transaction is chosen because it's least likely to be re-introduced into the mempool"

Reply:
Okay, I think my understanding of this was wrong. It should be "When the RecentlyEvictedCache is full, the transaction with the highest feeRate is removed to make room for the new transaction. This transaction is most likely to make it into the mempool, so by removing it, we allow it to be re-fetched."

Reply:
reworded